TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -200,17 +189,53 @@ private void runReplayWithIOStreams(Stream<TrafficStream> trafficChunkStream,
             var allRemainingWorkArray = requestToFinalWorkFuturesMap.values().toArray(CompletableFuture[]::new);
             log.info("All remaining work to wait on: " +
                     Arrays.stream(allRemainingWorkArray).map(cf->cf.toString()).collect(Collectors.joining()));
+            // remember, this block is ONLY for the leftover items.  Lots of other items have been processed
+            // and were removed from the live map (hopefully)
             CompletableFuture.allOf(allRemainingWorkArray)
-                    .whenComplete((t, v) -> {
+                    .handle((t, v) -> {
                         log.info("stopping packetHandlerFactory's group");
                         packetHandlerFactory.stopGroup();
+                        return null; // squash exceptions for individual requests
                     })
                     .get();
+            if (exceptionCount.get() > 0) {
+                log.warn(exceptionCount.get() + " requests to the target threw an exception; " +
+                        successCount.get() + " requests were successfully processed.");
+            } else {
+                log.info(successCount.get() + " requests were successfully processed.");
+            }
             assert requestToFinalWorkFuturesMap.size() == 0 :
                     "expected to wait for all of the in flight requests to fully flush and self destruct themselves";
         }
     }
 
+    private static AggregatedRawResponse packageAndWriteResponse(SourceTargetCaptureTuple.TupleToFileWriter tripleWriter, RequestResponsePacketPair rrPair, AggregatedRawResponse summary, Throwable t) {
+        log.warn("done sending and finalizing data to the packet handler");
+        SourceTargetCaptureTuple requestResponseTriple;
+        if (t != null) {
+            log.error("Got exception in CompletableFuture callback: ", t);
+            requestResponseTriple = new SourceTargetCaptureTuple(rrPair,
+                    new ArrayList<>(), Duration.ZERO
```

Comment:
Opening a ticket with Comparator so that it gracefully handles response times that are 0 https://opensearch.atlassian.net/browse/MIGRATIONS-1139